packages/amplify_api/test/graphql_helpers_test.dart

```diff
@@ -496,6 +497,153 @@ void main() {
         expect(req.decodePath, 'deleteBlog');
       });
     });
+
+    group('transformAppSyncJsonToModelJson', () {
+      test('should be no-op for flat JSON with no parents/children', () {
+        final input = <String, dynamic>{'id': 'abc123', 'name': 'Lorem Ipsum'};
+        final output = transformAppSyncJsonToModelJson(input, Blog.schema);
+        expect(input, output);
+      });
+
+      test('should be no-op for flat JSON with null parent/children', () {
+        final blogInput = <String, dynamic>{
+          'id': 'abc123',
+          'name': 'Lorem Ipsum',
+          'posts': null
+        };
+        final blogOutput =
+            transformAppSyncJsonToModelJson(blogInput, Blog.schema);
+        expect(blogInput, blogOutput);
+
+        final postInput = <String, dynamic>{
+          'id': 'abc123',
+          'title': 'Lorem Ipsum',
+          'rating': 0,
+          'comments': null,
+          'blog': null
+        };
+        final postOutput =
+            transformAppSyncJsonToModelJson(postInput, Post.schema);
+        expect(postInput, postOutput);
+      });
+
+      test('should translate children to expected format', () {
+        final input = <String, dynamic>{
+          'id': 'abc123',
+          'name': 'Lorem Ipsum',
+          'posts': {
+            'items': [
+              {'id': 'xyz456', 'rating': 0, 'title': 'Lorem ipsum'}
+            ]
+          }
+        };
+        final expectedOutput = <String, dynamic>{
+          'id': 'abc123',
+          'name': 'Lorem Ipsum',
+          'posts': [
+            {
+              'serializedData': {
+                'id': 'xyz456',
+                'rating': 0,
+                'title': 'Lorem ipsum'
+              }
+            }
+          ]
+        };
+        final output = transformAppSyncJsonToModelJson(input, Blog.schema);
+        expect(output, expectedOutput);
+      });
+
+      test('should translate parents and children to expected format', () {
+        final input = <String, dynamic>{
+          'id': 'xyz456',
+          'title': 'Lorem Ipsum',
+          'rating': 0,
+          'comments': {
+            'items': [
+              {'id': 'def456', 'content': 'Worst... post... ever!'}
+            ]
+          },
+          'blog': {'id': 'abc123', 'title': 'blog about life'}
+        };
+        final expectedOutput = <String, dynamic>{
+          'id': 'xyz456',
+          'title': 'Lorem Ipsum',
+          'rating': 0,
+          'comments': [
+            {
+              'serializedData': {
```

Comment:
Thanks for writing all of the code to translate between JSON formats.  Do you think it would have been better if we changed our own JSON instead to ensure total uniformity of JSON.

Reply:
long-term, aligning would be better, just don't have time right now to reconcile the datastore part and add extra codegen logic. This is the safest/easiest thing to do for now.